services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/northbound/switches/PortAntiflapSpec.groovy

```diff
@@ -105,4 +118,46 @@ timeout"() {
             islUtils.getIslInfo(isl).get().state == IslChangeType.DISCOVERED
         }
     }
+
+    /**
+     * In this test no actual port action is taken. We simulate port flapping by producing messages with OF events
+     * directly to kafka, because currently it is the only way to simulate an incredibly rapid port flapping that
+     * may sometimes occur on hardware switches(overheat?)
+     */
+    @Issue("https://github.com/telstra/open-kilda/issues/1827")
+    @Ignore("Due to https://github.com/telstra/open-kilda/issues/1827")
+    //there is a possible race condition that we exercise, so this needs couple reruns
+    @Rerun(times = 5)
+    def "System properly registers events order when port flaps incredibly fast"() {
+
+        when: "Port blinks rapidly for longer than 'antiflapWarmup' seconds, ending in UP state"
+        def producer = new KafkaProducer<>(producerProps)
+        def isl = topology.islsForActiveSwitches[0]
+        def messagesGenerated = 0
+        Wrappers.timedLoop(antiflapWarmup + 1) {
+            producer.send(new ProducerRecord(topoDiscoTopic, buildMessage(new PortInfoData(isl.srcSwitch.dpId,
+                    isl.srcPort, null, PortChangeType.DOWN)).toJson()))
+            producer.send(new ProducerRecord(topoDiscoTopic, buildMessage(new PortInfoData(isl.srcSwitch.dpId,
+                    isl.srcPort, null, PortChangeType.UP)).toJson()))
+            //note that we end with 'UP' event
+            messagesGenerated += 2
+            sleep(10) //do not overload storm, it can die
+        }
+        log.debug("Generated $messagesGenerated port up/down messages in ${antiflapWarmup + 1} seconds")
+
+        then: "Related ISL is FAILED"
+        Wrappers.wait(WAIT_OFFSET / 2.0) {
+            assert islUtils.getIslInfo(isl).get().state == IslChangeType.FAILED
+        }
+
+        and: "After the port cools down the ISL is discovered again"
+        TimeUnit.SECONDS.sleep(antiflapCooldown - 1)
```

Comment:
Could you please explain why we need to subtract -1?

Reply:
Also, why not move it to wait?

Reply:
It doesn't make sense to even start 'waiting' and disturbing northbound before the cooldown runs out. Subtracting `-1` to start actually 'waiting' right before the cooldown ends

Reply:
About -1. I thought so. Thanks.